### PR TITLE
pro-status: Use feature checks for VPScan and VPBackup

### PR DESCRIPTION
**Pull request by mreishus**
#### Changes proposed in this Pull Request:

* In pro-status, change VaultPress Scan and Backup checks to use feature checks instead of plan checks
  * Instead of using the logic that a free plan means they haven't purchased VaultPress scan and VaultPress backups, use the feature system to check if their plan provides those features
    * Our definition of `VAULTPRESS_BACKUPS` is wrong and should actually be `JETPACK_PERSONAL_AND_HIGHER`. This will need to be done in another PR. @obenland has seen another mention of this discrepancy as well.
      * Done here: D79837-code
      * Other reference is [here](https://github.com/Automattic/jetpack/pull/24121/files#r861998686)
  * Pro-status controls the top right of each card in the Jetpack Dashboard
    * For this section, it can display: Nothing, "Set Up" buttons inviting you to start using a feature you've purchased that you don't use, or a "Scan OK" or "Threats Detected" type of notice (None of which are in my screenshot).
  

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Currently hazy. Roughly, we will need to install/enable VaultPress (a standalone plugin) and make sure that on the top right of the cards in the Jetpack dashboard, you are invited to "Set Up" scan and backups if you have purchased them but have not yet enabled them. After you enable scan, the top right of the scan badge should show scan status.
  * If you have not purchased them, (you are on Jetpack free), then you should not see the SetUp buttons.
  * Behavior should be the same before and after.
  * I will be giving this a try and updating the PR as I find anything.

There may be another issue - not related - where the Jetpack Dashboard does not display properly when VaultPress is enabled. Some cards were always blinking in and out and my browser was making endless network requests. This should be handled in another PR if it is a real issue.
  * PR to fix this: #24194

### CODE REVIEW

**Comment by ebinnion** [comment]
@seear @vianasw - Can we get someone to give this a functional test?

**Comment by mreishus** [comment]
@singerb helped me out by using VPMC to move my site to VP Backup, and giving me a VP registration code which I entered. However, my site is stuck on the initial backup status (possibly because it's jetpack docker + JT?):

And my `getVaultPressData()` is still returning "N/A".
I haven't yet found a way to make `getVaultPressData()` return anything except "N/A" or a message about needing a VP registration code.

**Review by obenland — APPROVED** [review]

File: projects/plugins/jetpack/_inc/client/pro-status/index.jsx [hunk 1]
```diff
@@ -190,20 +191,19 @@ class ProStatus extends React.Component {
 			pluginSlug = 'akismet/akismet.php';
 		}
 
-		const hasPersonal = /jetpack_personal*/.test( sitePlan.product_slug ),
-			hasFree = /jetpack_free*/.test( sitePlan.product_slug ),
-			hasBackups = get( vpData, [ 'data', 'features', 'backups' ], false ),
-			hasVPScan = get( vpData, [ 'data', 'features', 'security' ], false );
-
-		const { scanStatus } = this.props;
+		const hasFree = /jetpack_free*/.test( sitePlan.product_slug ),
```
Comment (mmtr) [inline projects/plugins/jetpack/_inc/client/pro-status/index.jsx:194]: Is there any reason why this cannot be replaced with a feature check?

**Review by mmtr — COMMENTED** [review]
> Behavior should be the same before and after.

Everything worked as before for me, except this scenario:

- After installing and activating VaultPress (without entering any registration key), I purchased a Backups plan.
- Then, I went to Jetpack > Dashboard and I got different messages when compared to `master`:

`master` | This branch
--- | ---
Screen Shot 2022-05-04 at 16 09 50 | Screen Shot 2022-05-04 at 16 10 44

To me, it looks the behavior is wrong currently in `master` and this branch fixes it, but wanted to note that the behavior has changed, just in case we are expecting it to not change.

File: projects/plugins/jetpack/_inc/client/pro-status/index.jsx (hunk 1 above)
Comment (mmtr) [reply projects/plugins/jetpack/_inc/client/pro-status/index.jsx:194]: > We were going for smaller steps

That's fine, only wanted to double check whether we had any other restriction 🙂.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**